### PR TITLE
Add module settings to navigation

### PR DESCRIPTION
**Pull request by guillogo**
This PR will close issue 209

### CODE REVIEW

**Comment by guillogo** [comment]
Hi @safatshahin could you have a look to this PR please? Thank you.

File: lib/questionlib.php [hunk 1]
```diff
@@ -1764,17 +1764,23 @@ function question_edit_url($context) {
...
-function question_extend_settings_navigation(navigation_node $navigationnode, $context, $baseurl = '/question/edit.php') {
+function question_extend_settings_navigation(navigation_node $navigationnode, $context, $baseurl = '/question/edit.php', $default = true) {
     global $PAGE;
 
     if ($context->contextlevel == CONTEXT_COURSE) {
         $params = ['courseid' => $context->instanceid];
     } else if ($context->contextlevel == CONTEXT_MODULE) {
         $params = ['cmid' => $context->instanceid];
+        if ($default) {
+            $paramqbank = ['cmid' => $context->instanceid];
```
Comment (safatshahin) [inline lib/questionlib.php:1780]: if default is true, do we need this logic? its same as $param, so if its true, cant we use $param?

**Review by safatshahin — COMMENTED** [review]
Hi @guillogo 
I have made some comments in the code.
Also the seems the options are showing competency, filters etc, can you please have a look at those.
Thanks

File: lib/questionlib.php [hunk 2]
```diff
@@ -1764,17 +1764,23 @@ function question_edit_url($context) {
...
+ * @param context $context
  * @param string $baseurl the url of the base where the api is implemented from
+ * @param bool $default If use the default navigation or needs id as parameter.
  * @return navigation_node Returns the question branch that was added
  */
-function question_extend_settings_navigation(navigation_node $navigationnode, $context, $baseurl = '/question/edit.php') {
+function question_extend_settings_navigation(navigation_node $navigationnode, $context, $baseurl = '/question/edit.php', $default = true) {
     global $PAGE;
 
     if ($context->contextlevel == CONTEXT_COURSE) {
```
[the remark this answers is not in the document]
Comment (guillogo) [reply lib/questionlib.php:1775]: Agreed. probably at this point we don;t need it any more. Removed.

File: lib/questionlib.php (hunk 1 above)
Comment (guillogo) [reply lib/questionlib.php:1780]: As we need both, $paramqbank and $params (this last one for categories, export and import) so we can add the logic in each use of id or cmid depending of default or we can have it there and only do the validation once.

File: lib/questionlib.php [hunk 3]
```diff
@@ -1784,13 +1790,9 @@ function question_extend_settings_navigation(navigation_node $navigationnode, $c
     }
```
[the remark this answers is not in the document]
Comment (guillogo) [reply lib/questionlib.php:1790]: Thank you Safat for catching this one, adding the category parameter now to the new implementation.

**Comment by guillogo** [comment]
Hi @safatshahin
Changes done, can you please have a look. Also, about your question related with the items that are displayed in the list -> I'd say we can leave it as it is and when we have a decision what items should be there in issue 209 we can remove or add new things.
Thank you.

**Comment by safatshahin** [comment]
PR merged, thank you

**Pull request merged**